src/Plugin/GraphQL/DataProducer/Entity/EntityAccess.php

```diff
@@ -40,10 +41,12 @@ class EntityAccess extends DataProducerPluginBase {
    * @param string $operation
    * @param \Drupal\Core\Session\AccountInterface $user
    *
-   * @return bool|\Drupal\Core\Access\AccessResultInterface
+   * @return bool
    */
-  public function resolve(EntityInterface $entity, $operation = NULL, AccountInterface $user = NULL) {
-    return $entity->access($operation ?? 'view', $user);
+  public function resolve(EntityInterface $entity, ?string $operation, ?AccountInterface $user, FieldContext $context) {
```

Comment:
@klausi We're removing the ability to omit `operation` and `user` here (you'll now have to explicitly pass `NULL`). 

Questions: Does the context providing DataProducer handler always add those values? Do we consider these methods otherwise part of a public API that people might call directly?

Decision: Does that change constitute a breaking change?

Reply:
good point, this could be a breaking change. I think we should not remove the default values and keep the changes to a minimum. @dulnan can you fix that as well?

Reply:
Then we can't add the `FieldContext` though since adding a required parameter after optional parameters means the optional parameters aren't optional 🤔  so that slightly complicates things.

Reply:
Aha good point, then we need to keep it as proposed in this PR.

It should be fine though, the resolve call is mapped with all parameters, so should not matter in practice.

So the decision would be that this is not a breaking change.